openedx/core/djangoapps/credentials_service/models.py

```diff
@@ -1,3 +1,272 @@
 """
 Models for the credentials service.
 """
+import os
+import uuid  # pylint: disable=unused-import
+
+from django.conf import settings
+from django.contrib.sites.models import Site
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from model_utils.models import TimeStampedModel
+from xmodule_django.models import CourseKeyField
+
+
+MODE_CHOICES = (
+    (_('Honor'), 'honor'),
+    (_('Verified'), 'verified'),
+    (_('Professional'), 'professional'),
+)
+
+
+class AbstractCredential(TimeStampedModel):
+    """
+    Abstract Credentials configuration model.
+    """
+    site = models.ForeignKey(Site)
+    is_active = models.BooleanField(default=False)
+
+    def __unicode__(self):
```

Comment:
add meta for abstract class: https://docs.djangoproject.com/en/1.8/topics/db/models/#abstract-base-classes